src/Microsoft.AspNet.Server.Kestrel.Https/HttpsApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http.Features;
+using Microsoft.AspNet.Server.Kestrel.Filter;
+
+namespace Microsoft.AspNet.Server.Kestrel.Https
+{
+    public static class HttpsApplicationBuilderExtensions
+    {
+        public static IApplicationBuilder UseKestrelHttps(this IApplicationBuilder app, X509Certificate cert)
```

Comment:
This should take a full options class like how middleware work. There will be other options like TLS versions.